### ISSUE TITLE
NPE is thrown instead of a proper exception

### ISSUE THREAD

**Issue by jacool** [sksamuel/elastic4s#1367]
In some cases of server unavailability (I believe it was a 504 response) an NPE is thrown rather than some descriptive exception. (V6.2.5)

```
java.lang.NullPointerException: null
	at com.sksamuel.elastic4s.http.search.SearchImplicits$MultiSearchHttpExecutable$$anon$2.handle(SearchImplicits.scala:43)
	at com.sksamuel.elastic4s.http.search.SearchImplicits$MultiSearchHttpExecutable$$anon$2.handle(SearchImplicits.scala:39)
	at com.sksamuel.elastic4s.http.HttpClient$$anonfun$execute$1.apply(HttpClient.scala:69)
	at com.sksamuel.elastic4s.http.HttpClient$$anonfun$execute$1.apply(HttpClient.scala:68)
	at scala.util.Success$$anonfun$map$1.apply(Try.scala:237)
	at scala.util.Try$.apply(Try.scala:192)
	at scala.util.Success.map(Try.scala:237)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
	at scala.concurrent.Future$$anonfun$map$1.apply(Future.scala:237)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:36)
	at scala.concurrent.impl.ExecutionContextImpl$AdaptedForkJoinTask.exec(ExecutionContextImpl.scala:121)
	at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
	at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
	at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
	at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

**Comment by wajidp** [comment]
Any update on this?

**Comment by sksamuel** [comment]
I'll fix for 7.0 @wajidp

**Comment by sksamuel** [comment]
This is fixed in 7.0
